### ISSUE TITLE
[h3mlib] Comments miscalc: h3m_player.h

### ISSUE THREAD

**Issue by Dergash** [potmdehex/homm3tools#8]
Hello. Thanks for your work, you are awesome :)
I encountered a small miscalculation within h3m_player.h file. Comments says that H3M_PLAYER_EXT_WITH_HERO_ROE total size is 14 + hero name size. It's actually 13:

``` c
struct H3M_PLAYER_EXT_WITH_HERO_ROE     // Seen in The Mandate of Heaven (ROE)
{
    uint8_t starting_hero_is_random;
    uint8_t starting_hero_type; // CANNOT be 0xFF for this struct

    uint8_t starting_hero_face;
    uint32_t starting_hero_name_size;
    uint8_t starting_hero_name[];       // <starting_hero_name_size>
};                              // Total size 14+?(6+8+?), ? = <starting_hero_name_size>
```

6 bytes from fields within H3M_PLAYER_ROE, plus three mandatory uint8_t fields (3 bytes) and one uint32_t field (4 bytes). starting_hero_name[] not present when starting_hero_name_size is zero.
Same for:
H3M_PLAYER_EXT_WITH_TOWN_AND_HERO_ROE (16 total instead of 17).
H3M_PLAYER_EXT_WITH_HERO_ABSOD (14/15 instead of 15/16)

**Comment by potmdehex** [comment]
Thank you, nice find. Since you were the one who found this, you are free to make a pull request making you the contributor for it (size of contribution is irrelevant, this is not "too small"). If you don't want to I will correct it myself.

You are free to contact me if you have questions about h3mlib/the other code here, if you are interested in this stuff then I have much knowledge to share both on the code that I have written as well as the h3m format, the map editor and various versions of the HoMM 3 game.

**Comment by potmdehex** [comment]
Fixed by @Dergash: #9
